darshan-util/pydarshan/darshan/report.py

```diff
@@ -295,7 +297,8 @@ class DarshanReport(object):
 
 
     def __init__(self, 
-            filename=None, dtype='numpy', 
+            filename=None, 
+            dtype=None, # to be deprecated
```

Comment:
I think once you add the warning, as you have here, it is already considered "deprecated," so it is effectively "to be removed" at this point